src/Mod/CAM/Path/Machine/ui/editor/machine_editor.py

```diff
@@ -0,0 +1,1478 @@
+
+        # Show axes group if any axes configured
+        self.axes_group.setVisible(bool(linear_axes or rotary_axes))
+
+    def update_spindles(self):
+        """Update the spindle configuration UI based on spindle count.
+
+        Dynamically creates tabbed interface for multiple spindles with
+        input fields for name, ID, power, speed, and tool holder.
+        Updates Machine.spindles directly.
+        """
+        # Update machine spindles with current edits before rebuilding UI
+        if hasattr(self, "spindle_edits") and self.machine:
+            # Resize spindles list to match current edits
+            while len(self.machine.spindles) < len(self.spindle_edits):
+                self.machine.spindles.append(Spindle(name=""))
+            while len(self.machine.spindles) > len(self.spindle_edits):
+                self.machine.spindles.pop()
+
+            # Update each spindle from UI
+            for i, edits in enumerate(self.spindle_edits):
+                spindle = self.machine.spindles[i]
+                spindle.name = edits["name"].text()
+                spindle.id = edits["id"].text()
+                spindle.max_power_kw = edits["max_power_kw"].value()
+                spindle.max_rpm = edits["max_rpm"].value()
+                spindle.min_rpm = edits["min_rpm"].value()
+                spindle.tool_change = edits["tool_change"].itemData(
+                    edits["tool_change"].currentIndex()
+                )
+
+        # Clear existing spindle tabs
+        self.spindles_tabs.clear()
+        self.spindle_edits = []
+        count = self.spindle_count_combo.itemData(self.spindle_count_combo.currentIndex())
+
+        # Ensure machine has enough spindles
+        if self.machine:
+            while len(self.machine.spindles) < count:
+                self.machine.spindles.append(
+                    Spindle(
+                        name=f"Spindle {len(self.machine.spindles) + 1}",
+                        id=f"spindle{len(self.machine.spindles) + 1}",
+                        max_power_kw=3.0,
+                        max_rpm=24000,
+                        min_rpm=6000,
+                        tool_change="manual",
+                    )
+                )
+            while len(self.machine.spindles) > count:
+                self.machine.spindles.pop()
+
+        for i in range(count):
+            tab = QtGui.QWidget()
+            layout = QtGui.QFormLayout(tab)
+
+            # Get spindle object or use defaults
+            spindle = (
+                self.machine.spindles[i]
+                if self.machine and i < len(self.machine.spindles)
+                else None
+            )
+
+            name_edit = QtGui.QLineEdit()
+            name_edit.setText(spindle.name if spindle else f"Spindle {i+1}")
+            name_edit.textChanged.connect(
+                lambda text, idx=i: self._on_spindle_field_changed(idx, "name", text)
+            )
+            layout.addRow("Name:", name_edit)
```

Comment:
```suggestion
            layout.addRow("Name", name_edit)
```

Is this also translatable?